op-node/node/runtime_config.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/p2p"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+)
+
+var (
+	// UnsafeBlockSignerAddressSystemConfigStorageSlot is the storage slot identifier of the unsafeBlockSigner
```

Comment:
I'd like a test similar to `TestL1BlockSlots` in the following PR https://github.com/ethereum-optimism/optimism/commit/b232cf41278752acb8933a710d5265780479a669

This will ensure storage layout never changes, resulting in a bug. I do like the storage proof, having an `eth_call` would decouple the storage layout of the system config from the network's policy. Perhaps adding an `eth_call` and checking the value against the storage proof value, this would be a runtime check for the storage layout changing

Reply:
Or a better solution is to decouple to storage layout using a magic value

Reply:
We should consider using something like:

```solidity
bytes32(uint256(keccack256("systemconfig.unsafeblocksigner") - 1))
```

Also why is there the `- 1` ? Perhaps we shouldn't subtract one for simplicity

Reply:
👍 on the magic value. This will require a contract change but is much less brittle.